x-pack/test/functional/apps/security/doc_level_security_roles.ts

```diff
@@ -75,9 +76,12 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       const rowData = await PageObjects.discover.getDocTableIndex(1);
       expect(rowData).to.contain('EAST');
     });
+
     after('logout', async () => {
       // NOTE: Logout needs to happen before anything else to avoid flaky behavior
       await PageObjects.security.forceLogout();
+      await security.user.delete('userEast');
+      await security.role.delete('myroleEast');
```

Comment:
I was curious and looked at the code for deleting users/roles:

https://github.com/elastic/kibana/blob/808959e316082718c1c1081c362fbc6c91f2fdbc/test/common/services/security/user.ts#L34-L46

https://github.com/elastic/kibana/blob/0aabc317ec2a3d624c215c03ce472f236ae2b8bf/test/common/services/security/role.ts#L31-L44

Since we're doing this as a cleanup step, do you think we should wrap this in a try/catch to be safe?

Reply:
Do you propose to swallow any exceptions that might be thrown during test user/role deletion? 

I used to treat `after` as a fully fledged part of the test suite as it might affect subsequent tests. If subsequent test fails because `after` of another test failed, but was silenced it'd be much harder to debug what happened. Or what was your idea?

Reply:
That's a good point. I guess I was first getting tripped up on the difference between "delete role" (which doesn't fail on 404) and "delete user" (which does fail on 404), and I wasn't sure if there was a good reason for that distinction where it's used in other tests. 

So, yeah it's probably a bad idea to swallow _all_ errors, but do you think that "delete user" should swallow 404s too?